src/loki/polygon_search.cc

```diff
@@ -123,7 +150,8 @@ edges_in_rings(const google::protobuf::RepeatedPtrField<valhalla::Options_Ring>&
         auto edge_info = tile->edgeinfo(edge);
         bool intersects = false;
         for (const auto& ring_loc : bin.second) {
-          intersects = bg::intersects(rings_bg[ring_loc], edge_info.shape());
+          intersects = bg::intersects(rings_bg[ring_loc],
+                                      line_bg_t(edge_info.shape().begin(), edge_info.shape().end()));
```

Comment:
> avoid_polygons intersected edges as polygons instead of linestrings

so previously it was treating the edge shape as a polygon rather than a line? treating it as a line should be slightly cheaper but i dont understand how that would make the difference for having an intersection and not. maybe it has to do with those edge shapes making self intersecting rings and the self intersecting parts not counting for intersection?

Reply:
so the `ei.shape` has `std::vector<PointLL>` type which we register with bg as ring type so that's how it's treated in the bg::intersects call. But of course as an open ring which is the "undefined behavior" part for me.

Reply:
When I printed out the avoided edges from a polygon covering only one way I got back like 20 edges before this change. No idea what bg does internally in a situation like that. Now it only prints those 2 DE the polygon really intersects.